src/proto/FunctionRpc.proto

```diff
@@ -117,13 +117,35 @@ message WorkerInitRequest {
 
 // Worker responds with the result of initializing itself
 message WorkerInitResponse {
-  // Version of worker
+  // NOT USED
+  // TODO: Remove from protobuf during next breaking change release
   string worker_version = 1;
+
   // A map of worker supported features/capabilities
   map<string, string> capabilities = 2;
 
   // Status of the response
   StatusResult result = 3;
+
+  // Worker metadata captured for telemetry purposes
+  WorkerMetadata worker_metadata = 4;
+}
+
+message WorkerMetadata {
+  // The runtime/stack name
+  string runtime_name = 1;
+
+  // The version of the runtime/stack
+  string runtime_version = 2;
+
+  // The version of the worker or worker SDK
+  string worker_version = 3;
+
+  // The worker bitness/architecture
+  string worker_bitness = 4;
+
+  // Optional additional custom properties
+  map<string, string> custom_properties = 5;
```

Comment:
`optional map<string, string> custom_properties = 5;` ? 

Can you think of any other fields which shall be marked as optional?

Reply:
Just had a discussion about this in another PR! Turns out all properties are by default optional in proto v3. Also turns out it's not supported and so the build will fail with the error: 

`Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.`

Just removed the optional keyword in this last [PR](https://github.com/Azure/azure-functions-language-worker-protobuf/pull/77)

Reply:
Good to know. Thanks! I remember Madeline's PR where optional was used. I see that change was reverted.